vipe/oozie/graph.py

```diff
@@ -0,0 +1,284 @@
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__author__ = "Mateusz Kobos mkobos@icm.edu.pl"
+
+import yaml
+
+class OozieGraph(yaml.YAMLObject):
+    """A structure corresponding directly to Oozie XML file"""
+    yaml_tag = '!OozieGraph'
+    
+    def __init__(self, parameters, nodes):
+        """
+        Args:
+            parameters (Dict[string, string]): the key
+                is the name of parameter and the value is it's value.
+            nodes (Dict[string, Node]): a dictionary mapping name of a node 
+                to a Node object. The node with name 'start' is the one that 
+                the workflow starts from.
+        """
+        self.parameters = parameters
+        self.nodes = nodes
+    
+    def get_start(self):
+        """
+        Returns:
+            a Node that is the starting node of the workflow
+        """
+        return self.nodes['start']
+    
+    def __eq__(self, other):
+        return _default_eq(self, other)
+        
+
+def _default_eq(self, other):
+    """Default implementation of equality operator"""
+    return (type(other) is type(self)) and (self.__dict__ == other.__dict__)
+        
+class Node(yaml.YAMLObject):
+    yaml_tag = '!Node'
+
+class Action(Node):
+    def __init__(self, ok_node, error_node, configuration):
+        """
+        Args:
+            ok_node: name of Node object that is executed after this Action 
+                if no errors occurred during execution.
+            error_node: name of Node object that is executed after this Action 
+                if an error occurred during execution.
+            configuration (Dict[string, string]): the key is
+                the name of configuration property and the value is the
+                value of the configuration property.
+        """
+        self.ok_node = ok_node
+        self.error_node = error_node
+        self.configuration = configuration
+
+    def __eq__(self, other):
+        return _default_eq(self, other)
+
+class OtherAction(Action):
+    """Action that is not recognized by the program"""
+    yaml_tag = '!OtherAction'
+    
+    def __init__(self, ok_node, error_node, configuration, type_):
+        """
+        Args:
+            type_ (string): name of the XML tag corresponding to this action
+        """
+        super().__init__(ok_node, error_node, configuration)
+        self.type = type_
+
+    def __eq__(self, other):
+        return _default_eq(self, other)
+
+class SubworkflowAction(Action):
+    yaml_tag = '!SubworkflowAction'
+    
+    def __init__(self, ok_node, error_node, configuration, app_path):
```

Comment:
I wrote this comment before this one:

https://github.com/openaire/vipe/pull/12#issuecomment-151847933

so if it is not crucial from visualization POV we should be ok without including this one in processing.